tests/indexes_/test_checks.py

```diff
@@ -0,0 +1,205 @@
+from django.core import checks
+from django.db import models
+from django.test import TestCase, skipIfDBFeature, skipUnlessDBFeature
+from django.test.utils import (
+    isolate_apps,
+    override_system_checks,
+)
+
+from django_mongodb_backend.checks import check_vector_search_indexes
+from django_mongodb_backend.fields import ArrayField
+from django_mongodb_backend.indexes import SearchIndex, VectorSearchIndex
+
+
+@skipIfDBFeature("supports_atlas_search")
+@isolate_apps("indexes_", attr_name="apps")
+@override_system_checks([check_vector_search_indexes])
+class InvalidSearchIndexesTests(TestCase):
+    def test_requires_atlas_search_support(self):
+        class Article(models.Model):
+            title = models.CharField(max_length=10)
+
+            class Meta:
+                indexes = [
+                    SearchIndex(fields=["title"]),
+                ]
+
+        errors = checks.run_checks(app_configs=self.apps.get_app_configs(), databases={"default"})
+        self.assertEqual(
+            errors,
+            [
+                checks.Warning(
+                    "This MongoDB server does not support atlas search.",
+                    hint=(
+                        "The index won't be created. Silence this warning if you "
+                        "don't care about it."
+                    ),
+                    obj=Article._meta.indexes[0],
+                    id="django_mongodb_backend.indexes.SearchIndex.W001",
+                )
+            ],
+        )
+
+
+@isolate_apps("indexes_", attr_name="apps")
+@override_system_checks([check_vector_search_indexes])
+class InvalidVectorSearchIndexesTests(TestCase):
+    @skipIfDBFeature("supports_atlas_search")
```

Comment:
I don't like `skipUnlessDBFeature()` on every test. It would be simpler at the class level. I would move this test to `UnsupportedSearchIndexesTests` (renaming `InvalidSearchIndexesTests`) so you can do that.